### PR TITLE
State translation

### LINKED ISSUE #60: Allow translation of the entity `charge_state` and `plug_state`
Idea for improvement. You can add the following value in the string file to allow translation of the entity `charge_state` and `plug_state`:

```
_CHARGE_STATES = {
    0.0: ['Not charging', 'NOT_IN_CHARGE'],
    0.1: ['Waiting for planned charge', 'WAITING_FOR_PLANNED_CHARGE'],
    0.2: ['Charge ended', 'CHARGE_ENDED'],
    0.3: ['Waiting for current charge', 'WAITING_FOR_CURRENT_CHARGE'],
    0.4: ['Energy flap opened', 'ENERGY_FLAP_OPENED'],
    1.0: ['Charging', 'CHARGE_IN_PROGRESS'],
    # This next is more accurately "not charging" (<= ZE40) or "error" (ZE50).
    # But I don't want to include 'error' in the output text because people will
    # think that it's an error in Pyze when their ZE40 isn't plugged in...
    -1.0: ['Not charging or plugged in', 'CHARGE_ERROR'],
    -1.1: ['Not available', 'NOT_AVAILABLE']

_PLUG_STATES = {
    0: ['Unplugged', 'UNPLUGGED'],
    1: ['Plugged in', 'PLUGGED'],
    -1: ['Plug error', 'PLUG_ERROR'],
    -2147483648: ['Not available', 'NOT_AVAILABLE']
```

_Originally posted by @oncleben31 in https://github.com/epenet/hassRenaultZE/issues/30#issuecomment-711093434_

### PR DESCRIPTION
**Pull request by oncleben31**
Modification to allow translation of:
 - Charge and plug status sensors
 - Charge and plug binary sensors (need to have the [PR #4236](https://github.com/home-assistant/core/pull/42326) merge in HA core to work)

French translation added.

GitHub workflow to check the custom component with hassfest (mainly to check translation files)

Strings file added even if not used in custom component but could be used when merging this integration in HA core.

Fix #60

### CODE REVIEW

File: custom_components/renault/sensor.py [hunk 1]
```diff
@@ -190,7 +195,7 @@ def state(self):
                 plug_state = PlugState(data["plugStatus"])
             except ValueError:
                 plug_state = PlugState.NOT_AVAILABLE
-            return plug_state.name
+            return plug_state.name.lower().replace(" ", "_")
```
[the remark this answers is not in the document]
Comment (oncleben31) [reply custom_components/renault/sensor.py:198]: Thank you I've search for that helper without success.

**Comment by oncleben31** [comment]
I don't know why but when I add a file strings.sensor.json (to prepare for merge in HA) the translation stop working. So I remove the file.

File: .devcontainer/configuration.yaml [hunk 2]
```diff
@@ -6,3 +6,4 @@ logger:
     pyze: debug
     custom_components.renault: debug
 
+ptvsd:
```
Comment (epenet) [inline .devcontainer/configuration.yaml:9]: Can you open a separate PR for this, and explain why this is needed?

**Review by epenet — COMMENTED** [review]
Can you please rebase the PR?
I have cherry-picked a few of the changes, and it would easier to review the remaining changes if you would rebase it.

**Comment by oncleben31** [comment]
Rebased and tested in local. Translations work for status sensors.

File: .devcontainer/configuration.yaml (hunk 2 above)
Comment (oncleben31) [reply .devcontainer/configuration.yaml]: Deactivated with a comment . Is it OK ?

**Pull request merged**